crypto/sig/src/lib.rs

```diff
@@ -25,13 +26,9 @@ pub use schnorrkel::{Signature, SignatureError, SIGNATURE_LENGTH};
 /// Returns:
 /// * A 64-byte Schnorrkel Signature object which can be converted to and from bytes using its API.
 pub fn sign(context_tag: &[u8], private_key: &RistrettoPrivate, message: &[u8]) -> Signature {
-    // Nonce is hash( private_key || message )
-    // FIXME: We should probably hash the context_tag in as well.
-    // Or just use something like merlin instead of Blake2b256.
-    // In that case we make the assumption that Keccak, which underlies STROBE,
-    // is a strong Pseudorandom permutation, and that consequently Merlin with
-    // a partially secret input is a PRF.
+    // Nonce is hash( context_tag || private_key || message )
     let mut hasher = Blake2b256::new();
+    hasher.input(context_tag);
```

Comment:
I think it would be a good idea to add as a prefix, the length of the `context_tag` as well, since it is variable length
that prevents situations where `"", private_key1 "privatekey1 foo"` has the same nonce as a signature from another context`"privatekey1", privatekey1, "foo")`

Reply:
(i dont think we need framing around the message because it is the last thing in the sequence)

Reply:
i think this kind of framing is what merlin does: https://merlin.cool/transcript/ops.html#appending-messages

Reply:
I agree that this is a good idea. For simplicity's sake, you probably just want to use merlin again for the nonce generation here, using `Transcript.build_rng()`, so that you're getting identical behaviour w.r.t. domain separation as you are in the Schnorrkel signature (or, at least, no worse, since Schnorrkel is also using merlin).